internal/providers/info/info.go

```diff
@@ -0,0 +1,45 @@
+// Package info defines types used by the broker.
+package info
+
+// Group represents the group information that is fetched by the broker.
+type Group struct {
+	Name string `json:"name"`
+	UGID string `json:"ugid"`
+}
+
+// User represents the user information obtained from the provider.
+type User struct {
+	Name   string  `json:"name"`
+	UUID   string  `json:"uuid"`
+	Home   string  `json:"dir"`
+	Shell  string  `json:"shell"`
+	Gecos  string  `json:"gecos"`
+	Groups []Group `json:"groups"`
+}
+
+// NewDefaultUser creates a new user with default values.
+func NewDefaultUser(name string) User {
+	return NewUser(name, "", "", "", "", nil)
+}
+
+// NewUser creates a new user with the specified values.
+//
+// It fills the defaults for Shell and Gecos if they are empty.
+func NewUser(name, home, uuid, shell, gecos string, groups []Group) User {
```

Comment:
I would still have tests for them. Please ensure that all public API for each package is tested. We have already seen places where you skipped the testNew(), then New() becomes more sofisticated and I need to implement tests for those…